osu.Game.Rulesets.Mania/Mods/ManiaModFadeIn.cs

```diff
@@ -19,5 +30,103 @@ public class ManiaModFadeIn : Mod
+
+            foreach (Column column in maniaPlayfield.Stages.SelectMany(stage => stage.Columns))
+            {
+                HitObjectContainer hoc = column.HitObjectArea.HitObjectContainer;
+                Container hocParent = (Container)hoc.Parent;
+
+                hocParent.Remove(hoc);
+                hocParent.Add(new BufferedContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        hoc,
+                        new LaneCover(0.5f, false)
+                        {
+                            RelativeSizeAxes = Axes.Both
+                        }
+                    }
+                });
+            }
+        }
+
+        private class LaneCover : CompositeDrawable
+        {
+            private readonly Box gradient;
+            private readonly Box filled;
+
+            public LaneCover(float initialCoverage, bool reversed)
+            {
+                Blending = new BlendingParameters
+                {
+                    RGBEquation = BlendingEquation.Add,
+                    Source = BlendingType.Zero,
+                    Destination = BlendingType.One,
+                    AlphaEquation = BlendingEquation.Add,
+                    SourceAlpha = BlendingType.Zero,
+                    DestinationAlpha = BlendingType.OneMinusSrcAlpha
+                };
+                InternalChildren = new Drawable[]
+                {
+                    gradient = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        RelativePositionAxes = Axes.Both,
+                        Height = 0.25f
+                    },
+                    filled = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both
+                    }
+                };
+                Coverage = initialCoverage;
+                Reversed = reversed;
+            }
+
+            private float coverage;
+
+            public float Coverage
+            {
+                set
+                {
+                    filled.Height = value;
+                    gradient.Y = reversed ? 1 - value - gradient.Height : value;
+                    coverage = value;
+                }
+            }
+
+            private bool reversed;
+
+            public bool Reversed
+            {
+                set
+                {
+                    filled.Anchor = value ? Anchor.BottomLeft : Anchor.TopLeft;
+                    filled.Origin = value ? Anchor.BottomLeft : Anchor.TopLeft;
+                    gradient.Colour = ColourInfo.GradientVertical(
+                        Color4.White.Opacity(value ? 0f : 1f),
+                        Color4.White.Opacity(value ? 1f : 0f)
+                    );
+
+                    reversed = value;
+                    Coverage = coverage; //re-apply coverage to update visuals
```

Comment:
We don't usually do this. Make an `updateCoverage()` method, and implement `Coverage` something like:
```csharp
private float coverage;

public float Coverage
{
    get => coverage;
    set
    {
        if (coverage == value)
            return;

        coverage = value;
        filled.Height = ...
        ...

        updateCoverage()
    }
}
```